x/launch/keeper/msg_server_settle_request.go

```diff
@@ -0,0 +1,128 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	spnerrors "github.com/tendermint/spn/pkg/errors"
+	"github.com/tendermint/spn/x/launch/types"
+)
+
+func (k msgServer) SettleRequest(
+	goCtx context.Context,
+	msg *types.MsgSettleRequest,
+) (*types.MsgSettleRequestResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	chain, found := k.GetChain(ctx, msg.ChainID)
+	if !found {
+		return nil, sdkerrors.Wrap(types.ErrChainNotFound, msg.ChainID)
+	}
+
+	if chain.LaunchTriggered {
+		return nil, sdkerrors.Wrap(types.ErrTriggeredLaunch, msg.ChainID)
+	}
+
+	coordAddress := k.profileKeeper.GetCoordinatorAddressFromID(ctx, chain.CoordinatorID)
+	if msg.Coordinator != coordAddress {
+		return nil, sdkerrors.Wrap(types.ErrNoAddressPermission, msg.Coordinator)
+	}
+
+	// first check if the request exist
+	request, found := k.GetRequest(ctx, msg.ChainID, msg.RequestID)
+	if !found {
+		return nil, sdkerrors.Wrapf(types.ErrRequestNotFound,
+			"request %d for chain %s not found",
+			msg.RequestID,
+			msg.ChainID,
+		)
+	}
+
+	// perform request action
+	k.RemoveRequest(ctx, msg.ChainID, request.RequestID)
+	if msg.Approve {
+		err := applyRequest(ctx, k.Keeper, msg, request)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &types.MsgSettleRequestResponse{}, nil
+}
+
+// applyRequest approve the request and perform
```

Comment:
```suggestion
// applyRequest approves the request and performs
```

Reply:
Can we move this method to `request.go` since it will not only be called by this specific message